src/main/java/org/galatea/starter/service/IexClient.java

```diff
@@ -30,6 +32,22 @@
    * @return a list of the last traded price for each of the symbols passed in.
    */
   @GetMapping("/tops/last")
-  List<IexLastTradedPrice> getLastTradedPriceForSymbols(@RequestParam("symbols") String[] symbols);
+  List<IexLastTradedPrice> getLastTradedPriceForSymbols(@RequestParam("symbols") String[] symbols,
+      @RequestParam("token") String token);
+
+  /**
+   * Get historical price data (close, high, low, open, and volume) for the given symbol
+   * over a specified time range. See https://iexcloud.io/docs/api/#historical-prices.
+   * @param symbol A string representing a stock symbol for which to retrieve historical data.
+   * @param range A string specifying a range and date. See link.
+   * @return A list of IexHistoricalPrice objects for the symbol for each date in the range of time.
+   */
+
+  @GetMapping("/stock/{symbol}/chart/{rangeAndDate}")
```

Comment:
Why are combining rangeAndDate here?

Reply:
Resolved this issue by splitting getHistoricalPrices method of IexClient: one for the case when range="date" and one for otherwise.